### PR TITLE
Migrate MIO communication to pibridge library

### PR DESCRIPTION
**Pull request by l1k**
... and throw in a bunch of cleanups for good measure.

All of these changes have only been compile-tested so far. I don't have a MIO module at my disposal for testing. If someone could test this on actual hardware I'd be grateful. Thanks!

### CODE REVIEW

**Comment by l1k** [comment]
Depends on https://github.com/RevolutionPi/linux/pull/135 to compile cleanly.

**Comment by nbuchwitz** [comment]
Successfully tested in mixed setup (arm64):

```
{
     "time_local": "2022-12-01T12:12:34.435184",
     "duration_s": 30.0,
     "cycles": 3000,
     "min_ms": 22,
     "max_ms": 23,
     "mean_ms": 22.82,
     "module_config": [
          "device_RevPiDO_20160818_1_0_001",
          "device_RevPiDI_20160818_1_0_002",
          "device_RevPiMIO_20200901_1_0_001",
          "device_RevPiAIO_20170301_1_0_002",
          "device_RevPiCore_20170404_1_2_001",
          "device_RevPiAIO_20170301_1_0_001",
          "device_RevPiDIO_20160818_1_0_001",
          "device_RevPiDIO_20160818_1_0_002",
          "device_RevPiDI_20160818_1_0_001"
     ],
     "system": {
          "kernel_version": "#1 SMP PREEMPT_RT Thu, 01 Dec 2022 10:50:13 +0100",
          "kernel_release": "5.10.152-rt75-v8+",
          "machine": "aarch64"
     }
}
```

Later today I plan to test with armhf too.

**Comment by linosanfilippo-kunbus** [comment]
Code changes look good, but I see a lot of error when using the MIO:

```
 pi-bridge serial0-0: receive head error in io-req
 piControl: addr 31: io communication failed with AIO -5
 pi-bridge serial0-0: invalid checksum (expected: 0x02, got 0x17)
```
Those error do not result from these changes though, the problems seems to be 2b3dd1a28b3e931b77279163ff6ea0e27e77af75 ("pibridge-serdev: fix wait for event"). I am currently investigating this and will wait with my approvement for this PR until I have verified that it does not break anything else.

**Review by linosanfilippo-kunbus — APPROVED** [review]

File: revpi_mio.c [hunk 1]
```diff
@@ -160,12 +156,12 @@ static unsigned int revpi_chnl_compress(void *dst, void *src,
 					unsigned long bitmap, int step)
 {
 	unsigned char *d, *s;
-	unsigned int i = 0;
+	unsigned int i, last = __fls(bitmap);
 
 	s = (unsigned char *) src;
 	d = (unsigned char *) dst;
 
-	for (i = 0; i < 32 && VALUE_MASK_H(bitmap, i); i++) {
```
[the remark this answers is not in the document]
Comment (l1k) [reply revpi_mio.c:177]: <blockquote>
AFAIU in the original version the next loop iteration is only processed if there is at least one more bit set in the bitmap.
</blockquote>

Right.

<blockquote>
 In the new version the iteration starts at the highest bit but then iterates until it reaches bit 0 no matter if any other bit beside the highest set bit is set.
</blockquote>

Huh? The iteration starts at 0 and ends at the highest set bit. So functionally this should be equivalent to the previous code, but (hopefully) easier to read and more performant.

**Comment by l1k** [comment]
Glad to hear that, thanks for the review and tests @linosanfilippo-kunbus @nbuchwitz  !

**Pull request merged**